### ISSUE TITLE
Flutter_compass issue in apk build

### ISSUE THREAD

**Issue by md-lotfi** [tlserver/flutter_map_location_marker#131]
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':flutter_compass'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
Running Gradle task 'assembleRelease'...                         1,569ms
Gradle task assembleRelease failed with exit code 1

**Comment by vaghe777** [comment]
Same issue Here!

**Comment by boranmohamed** [comment]
same here

**Comment by dariocavada** [comment]
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':flutter_compass'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.


with the latest flutter version:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.6.1 23G93 darwin-arm64, locale it-IT)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.93.1)
[✓] Connected device (6 available)
[✓] Network resources

**Comment by dariocavada** [comment]
If you need a solution use this in your pubspec.yaml:

``` yaml
 flutter_map_location_marker:
    git:
      https://github.com/dariocavada/flutter_map_location_marker
```

I create this fork with the updated compass ([dchs_flutter_compass](https://pub.dev/packages/dchs_flutter_compass))

**Comment by Landjager** [comment]
that fixxed it for me. Thanks!

> If you need a solution use this in your pubspec.yaml:
> 
> ```yaml
>  flutter_map_location_marker:
>     git:
>       https://github.com/dariocavada/flutter_map_location_marker
> ```
> 
> I create this fork with the updated compass ([dchs_flutter_compass](https://pub.dev/packages/dchs_flutter_compass))

That fixxed it for me. Thanks!

**Comment by tlserver** [comment]
For anyone facing this issue, consider trying `v10.0.0-beta.0`. In this new version, FMLM has dropped the `flutter_compass` dependency and now uses `flutter_rotation_sensor`.

The `flutter_rotation_sensor` offers two great features: customizable update intervals and coordinate transformation. Additionally, you can use any device heading provider; `flutter_rotation_sensor` is just the default implementation. Check out the custom stream example to learn how to set it up.

**Comment by TimBaumgart** [comment]
> For anyone facing this issue, consider trying `v10.0.0-beta.0`. In this new version, FMLM has dropped the `flutter_compass` dependency and now uses `flutter_rotation_sensor`.
> 
> The `flutter_rotation_sensor` offers two great features: customizable update intervals and coordinate transformation. Additionally, you can use any device heading provider; `flutter_rotation_sensor` is just the default implementation. Check out the custom stream example to learn how to set it up.

This creates a new error for me when trying to build the ios app.

```
Swift Compiler Error (Xcode): Cannot find 'RotationSensorPlugin' in scope
/Users/builder/.pub-cache/hosted/pub.dev/flutter_rotation_sensor-0.1.0/ios/Classes/FlutterRotationSensorPlugin.swift:14:19

Swift Compiler Error (Xcode): Value of type 'CMDeviceMotion' has no member 'orientation'
/Users/builder/.pub-cache/hosted/pub.dev/flutter_rotation_sensor-0.1.0/ios/Classes/FlutterRotationSensorPlugin.swift:45:35
```

Any ideas?

**Comment by Landjager** [comment]
> > For anyone facing this issue, consider trying `v10.0.0-beta.0`. In this new version, FMLM has dropped the `flutter_compass` dependency and now uses `flutter_rotation_sensor`.
> > The `flutter_rotation_sensor` offers two great features: customizable update intervals and coordinate transformation. Additionally, you can use any device heading provider; `flutter_rotation_sensor` is just the default implementation. Check out the custom stream example to learn how to set it up.
> 
> This creates a new error for me when trying to build the ios app.
> 
> ```
> Swift Compiler Error (Xcode): Cannot find 'RotationSensorPlugin' in scope
> /Users/builder/.pub-cache/hosted/pub.dev/flutter_rotation_sensor-0.1.0/ios/Classes/FlutterRotationSensorPlugin.swift:14:19
> 
> Swift Compiler Error (Xcode): Value of type 'CMDeviceMotion' has no member 'orientation'
> /Users/builder/.pub-cache/hosted/pub.dev/flutter_rotation_sensor-0.1.0/ios/Classes/FlutterRotationSensorPlugin.swift:45:35
> ```
> 
> Any ideas?

I'm assuming you have imported the dependency? If not try `flutter pub add  flutter_rotation_sensor:0.1.0`. Else can you show some code where the error occurs?

**Comment by tlserver** [comment]
flutter_rotation_sensor is updated to v0.1.1. The build issue of ios app should be fixed.

**Comment by jiazeh** [comment]
Waiting for `flutter_rotation_sensor v0.1.1` to be merged into `flutter_map_location_marker: ^10.0.0-beta.1`
